rm-datasource/src/main/java/io/seata/rm/datasource/undo/AbstractUndoExecutor.java

```diff
@@ -145,9 +164,115 @@ protected void undoPrepare(PreparedStatement undoPST, ArrayList<Field> undoValue
      * Data validation.
      *
      * @param conn the conn
-     * @throws SQLException the sql exception
+     * @return return true if data validation is ok and need continue
+     * @throws SQLException the sql exception such as has dirty data
      */
-    protected void dataValidation(Connection conn) throws SQLException {
+    protected boolean dataValidation(Connection conn) throws SQLException {
+        
```

Comment:
It ‘s better to add a switcher to close  the dirty data check manually, the switcher is open default.

Reply:
OK, I will add it in the next standalone PR.